oneoffs/opening_freqs_export.py

```diff
@@ -190,39 +192,106 @@ def create_hourly_reports(hour_directory):
 """
 
 
+def get_runs(db):
+    return [r[0] for r in db.execute('''
+               select distinct(run) from joseki_counts group by 1'''
+                                     ).fetchall()]
+
+def top_seqs_by_run(db, topN=300):
+    runs = get_runs(db)
+    top_seqs_by_run = {}
+    for r in runs:
+        cur = db.execute('''
+                         select seq, sum(count) from joseki_counts
+                         where run = ?
+                         group by seq order by 2 desc limit ?;
+                         ''', (r, topN))
+        sequences = [row for row in cur.fetchall()]
+        sequences = collapse_common_prefix(sequences)
+        print("Pruned to ", len(sequences))
+        sequences.sort(key=lambda s: s[1], reverse=True)
+        top_seqs_by_run[r] = sequences
+    return top_seqs_by_run
+
+
+def run_time_ranges(db):
+    ts = lambda hr: int(dt.datetime.strptime(hr, "%Y-%m-%d-%H").timestamp())
+    runs = [ (r[0], (ts(r[1]), ts(r[2]))) for r in db.execute('''
+        select run, min(hour), max(hour) from joseki_counts group by 1;
+        ''').fetchall()]
+    return dict(runs)
```

Comment:
{r[0]: (ts(r[1]), ts(r[2])) ....}

or

{r: (ts(min_t), ts(max_t)) for r, (min_t, max_t) in db.execute...

Reply:
Done.